spec/formatting.md

```diff
@@ -936,7 +936,7 @@ The _Default Bidi Strategy_ is defined as follows:
    right-to-left directionality, and to the message's directionality not being known.
 1. For each _expression_ `exp` in _pattern_:
    1. Let `fmt` be the formatted string representation of the _resolved value_ of `exp`.
-   1. Let `dir` be the directionality of `fmt`,
+   1. Let `dir` be `DIR(exp)`,
```

Comment:
The direction cannot be defined only from `exp`, as its _resolved value_ also depends on the formatting context.

I'm not convinced that defining `DIR()` makes this algorithm clearer, as in the immediate context we're also reading "the formatted string representation of the _resolved value_ of `exp`" as well as "the `u:dir` _option_ of the _resolved value_ of `exp`"

Something like this might be better:
```
   1. Let `resval` be the _resolved value_ of `exp`.
   1. Let `fmt` be the formatted string representation of `resval`.
   1. Let `dir` be the directionality of `resval`,
      one of « `'LTR'`, `'RTL'`, `'unknown'` », with the same meanings as for `msgdir`.
   1. Let the boolean value `isolate` be
      True if the `u:dir` _option_ of `resval` has a value other than `'inherit'`,
      or False otherwise.
```

And then reactoring the "The auxiliary function `DIR`..." paragraph into a non-normative note.